src/test/java/dev/openfeature/sdk/OpenFeatureClientTest.java

```diff
@@ -42,7 +42,7 @@ void reset_logs() {
     @DisplayName("should not throw exception if hook has different type argument than hookContext")
     void shouldNotThrowExceptionIfHookHasDifferentTypeArgumentThanHookContext() {
         OpenFeatureAPI api = OpenFeatureAPI.getInstance();
-        api.setProvider("shouldNotThrowExceptionIfHookHasDifferentTypeArgumentThanHookContext", new DoSomethingProvider());
+        api.setProviderAndWait("shouldNotThrowExceptionIfHookHasDifferentTypeArgumentThanHookContext", new DoSomethingProvider());
```

Comment:
This test isn't helped by using the non-blocking provider setter here. There are a few tests where we want to not block (to assert behaviors about the SDK before the provider is ready, for instance) but this is not one.